packages/lib/event-types/transformers/api-to-internal/locations.ts

```diff
@@ -0,0 +1,65 @@
+import { z } from "zod";
+
+import { type CreateEventTypeInput_2024_06_14, type Integration_2024_06_14 } from "@calcom/platform-types";
+
+const integrationsMapping: Record<Integration_2024_06_14, string> = {
+  "cal-video": "integrations:daily",
+};
+
+export function transformLocationsApiToInternal(
+  inputLocations: CreateEventTypeInput_2024_06_14["locations"]
+) {
+  if (!inputLocations) {
+    return [];
+  }
+
+  return inputLocations.map((location) => {
+    const { type } = location;
+    switch (type) {
+      case "address":
+        return { type: "inPerson", address: location.address, displayLocationPublicly: location.public };
+      case "link":
+        return { type: "link", link: location.link, displayLocationPublicly: location.public };
+      case "integration":
+        const integrationLabel = integrationsMapping[location.integration];
+        return { type: integrationLabel };
+      case "phone":
+        return {
+          type: "userPhone",
+          hostPhoneNumber: location.phone,
+          displayLocationPublicly: location.public,
+        };
```

Comment:
can we infer the type of UserPhoneSchema and here write 

```ts
case "phone":
        return {
          type: "userPhone",
          hostPhoneNumber: location.phone,
          displayLocationPublicly: location.public,
        } satisfies UserPhoneLocationType;
 ```

Reply:
Done in [refactor: satisfies check for event-type locations](https://github.com/calcom/cal.com/pull/16685/commits/6be1db3cac8ec2606564c057baf5a11230c82224)